### PR TITLE
Feature #122642 feat: New Field type - UCM subform

### PR DESCRIPTION
**Pull request by komalm**

### CODE REVIEW

File: administrator/models/fields/formsourcefield.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * @version    SVN: <svn_id>
+ * @package    TJ-Fields
+ * @author     Techjoomla <extensions@techjoomla.com>
+ * @copyright  Copyright (c) 2009-2016 TechJoomla. All rights reserved.
+ * @license    GNU General Public License version 2 or later.
+ */
+
+defined('JPATH_BASE') or die;
```
Comment (ankush-maherwal) [inline administrator/models/fields/formsourcefield.php]: defined('JPATH_PLATFORM') or die;

File: administrator/models/fields/formsourcefield.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ * @since  1.6
+ */
+class JFormFieldformsourcefield extends JFormField
+{
+	/**
+	 * The form field type.
+	 *
+	 * @var		string
+	 *
+	 * @since	1.6
```
Comment (ankush-maherwal) [inline administrator/models/fields/formsourcefield.php]: check version

File: administrator/models/fields/formsourcefield.php [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+	 */
+	protected function getInput()
+	{
+		$db    = JFactory::getDbo();
+		$query = $db->getQuery(true);
+
+		$query->select($db->quoteName(array('id', 'title', 'unique_identifier')));
+		$query->from($db->quoteName('#__tj_ucm_types'));
```
Comment (ankush-maherwal) [inline administrator/models/fields/formsourcefield.php]: check if UCM is installed or not

File: administrator/models/fields/formsourcefield.php [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+	 * @param   string  $name          Field name
+	 * @param   string  $value         Field value
+	 * @param   string  &$node         Field element
+	 * @param   string  $control_name  Field control_name
+	 *
+	 * @return   list  The list of form source.
+	 *
+	 * @since   1.6
+	 */
+	public function fetchformsource($name, $value, &$node, $control_name)
```
Comment (ankush-maherwal) [inline administrator/models/fields/formsourcefield.php]: empty function?

Do we need this?

File: administrator/models/fields/formsourcefield.php (hunk 1 above)
Comment (komalm) [reply administrator/models/fields/formsourcefield.php]: Done

File: administrator/models/fields/formsourcefield.php (hunk 2 above)
Comment (komalm) [reply administrator/models/fields/formsourcefield.php]: 1.3 update

File: administrator/models/fields/formsourcefield.php (hunk 4 above)
Comment (komalm) [reply administrator/models/fields/formsourcefield.php]: remove

File: administrator/models/fields/formsourcefield.php (hunk 3 above)
Comment (komalm) [reply administrator/models/fields/formsourcefield.php]: Done.
create new field type "tjfieldfields" this type return all fields which are support by tjfields depending upon the components installed.

"Ucm subform" field depends upon com_tjucm. if ucm is not installed the "Ucm Subform" field should not display in the field list.

File: administrator/models/fields/layoutfield.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	 * @since	1.3
+	 */
+	protected $type = 'text';
+
+	/**
+	 * Method to get the field input markup.
+	 *
+	 * @return	string	The field input markup.
+	 *
+	 * @since	1.6
```
Comment (ankush-maherwal) [inline administrator/models/fields/layoutfield.php]: version

**Pull request merged**